vercel/speed-insights
# feat: Astro support

### 📓 What's in there?

This is an initial draft for Astro first-class support.

`beforeSend` is supported by using  `speedInsightsBeforeSend` as global window function
```
<script is:inline>
  function speedInsightsBeforeSend(data){
    console.log("Capured event!")
    return data;
  }
</script>
```

TODO list:
- [ ] use astro@4
- [ ] check types and autocompletion

### 🧪 How to test?

There's a new example app, based on the Astro official blog template
- get the dependencies `pnpm i`
- update `apps/astro/src/components/BaseHead.astro` to set `scriptSrc="https://va.vercel-scripts.com/v1/speed-insights/script.js"` on the speed insights component. This will attempt to report metrics so we can see the reported payloads.
- `cd apps/astro && pnpm dev` to start the app
- visit the app, and navigate its pages.

### ❗ Notes to reviewers
